build/ProjectReunion-CI.yml

```diff
@@ -37,10 +38,16 @@ jobs:
   - template: AzurePipelinesTemplates\ProjectReunion-BuildDevProject-Steps.yml
   - template: AzurePipelinesTemplates\ProjectReunion-PublishProjectOutput-Steps.yml
 
+# MRTCore build pipeline
+- template: ..\dev\MRTCore\azure-pipelines.yml
+  parameters:
+    buildJobName: 'BuildMRTCore'
+    
 - template: AzurePipelinesTemplates\ProjectReunion-RunHelixTests-Job.yml
   parameters:
     name: 'RunTestsInHelix'
-    dependsOn: Build
+    dependsOn: 
+      - Build
```

Comment:
>Build [](start = 8, length = 5)

MRTCore is not part of Build

Reply:
Sure, but in the future as we start to add more tests, projects, etc,  I will imagine we will want our tests to wait on the build of the projectreunion.dll

Reply:
yes, it should depend on ALL build jobs (which have tests)

---
In reply to: [520097813](https://github.com/microsoft/ProjectReunion/pull/282#discussion_r520097813) [](ancestors = 520097813)

Reply:
MRTCore takes ~15 minutes to build and we can build our tests separately from it, they do not depend on that finishing and don't use any of the binaries from that job because we specifically skip over the test binaries.  

I rather get the tests out to the helix queues asap. Realistically we can remove the depends on main build but we will have to put it back eventually.

Reply:
how do you run MRTCore tests without waiting for MRTCore finished to build?

---
In reply to: [520168205](https://github.com/microsoft/ProjectReunion/pull/282#discussion_r520168205) [](ancestors = 520168205)

Reply:
Sorry I am getting lost between the email comments and the ones built in here... I guess they are not compatible.

When I was originally looking at the build MRTCore pipeline I thought I saw some logs that said something like "BaseUnitTests are not built in this configuration" Maybe I was just looking at the wrong flavor? In the test pipeline we build BaseUnitTests ourselves and then send that and their dependencies over to the helix queue.  If that is being built in the MRTCore pipeline we can just grab those artifacts that get published and depend on it finishing instead.

Reply:
So looking at this further, it is definitely more efficient to just split these up regardless. It will save on pipeline runtime if we can build MRTCore and the tests in parallel.  That way we can start running the tests on helix as soon as they are built (3 minute build time) instead of waiting for all of MRTCore to build (15 minute build time). 

I will defer to you what you think is best.